bms-monitor.js

```diff
@@ -1,18 +1,83 @@
-const {expect} = require('chai');
-
-function batteryIsOk(temperature, soc, charge_rate) {
-    if (temperature < 0 || temperature > 45) {
-        console.log('Temperature is out of range!');
-        return false;
-    } else if (soc < 20 || soc > 80) {
-        console.log('State of Charge is out of range!')
-        return false;
-    } else if (charge_rate > 0.8) {
-        console.log('Charge rate is out of range!');
-        return false;
-    }
-    return true;
-}
-
-expect(batteryIsOk(25, 70, 0.7)).to.be.true;
-expect(batteryIsOk(50, 85, 0)).to.be.false;
+// Setting the defautl language English
+let LANGUAGE = 'en';
+
+const PARAMETER_LIMITS = {
+  temperature: {min: 0, max: 45},
+  soc: {min: 20, max: 80},
+  chargeRate: {min: 0, max: 0.8},
+};
+
+const WARNING_TOLERANCE = 0.05;
+
+const WARNING_MESSAGES = {
+  en: {
+    temperature: {
+      low: 'Warning: Approaching lower temperature limit!',
+      high: 'Warning: Approaching upper temperature limit!',
+      error: 'Temperature is out of range!',
+    },
+    soc: {
+      low: 'Warning: Approaching discharge!',
+      high: 'Warning: Approaching charge-peak!',
+      error: 'State of Charge is out of range!',
+    },
+    chargeRate: {
+      low: 'Warning: Approaching minimum charge rate!',
+      high: 'Warning: Approaching maximum charge rate!',
+      error: 'Charge rate is out of range!',
+    },
+  },
+  de: {
```

Comment:
Good mapping of the messages in two languages. But the messages are not tested?
Consider this: Someone asks to add a parameter and we do that. The developer adds it under the `en:` but forgets to update the `de:` object. Think of a test that can catch this mistake, without having to test every situation in every language.